include/OnnxMlirRuntime.h

```diff
@@ -123,13 +146,21 @@
  *
  * \subsection freeing-tensor-memory Freeing Tensor Memory
  *
- *
- * When creating an OMTensor with a specified tensor array, a user has the option to
- * create a tensor with or without ownership (omTensorCreateWithOwnership).
- *
- * If the ownership flag is set to "false", then a user is responsible for freeing memory until the last use.
- * Otherwise, if the flag is set to "true", then the destruction of the tensor (omTensorDestroy)
- * will also free any associated memory and data buffers.
+ * In general, if a caller creates a tensor object (omTensorCreate), they are
+ * responsible for deallocating the data buffer separately after the tensor is
+ * destroyed. If onnx-mlir creates the tensor (run_main_graph), then the
+ * tensor object owns the data buffer and its freed automatically when the
+ * tensor is destroyed.
+ *
+ * This default behavior can be changed. When creating a tensor, a user may use
+ * omTensorCreateWithOwnership to explicitly set data buffer ownership. Otherwise,
+ * after a tenor is created, omTensorSetOwning can be used to change
+ * the ownership setting.
+ *
+ * When omTensorDestroy is called, if the ownership flag is set to "true",
+ * then the destruction of the tensor will also free any associated data buffer
+ * memory. If the ownership flag is set to "false", then the user is responsible
+ * for freeing the data buffer memory after destroying the tensor.
```

Comment:
Need any clarification on OMTensorList?

Reply:
Based on the subsection heading I was focused on Tensor objects. However I'd agree a summary on how TensorLists are freed would be useful. It's related enough that it make sense to include under this heading. I'll add a few sentences on the next commit.